src/Plugin/RetryPlugin.php

```diff
@@ -35,18 +45,30 @@
      * @param array $config {
      *
      *     @var int $retries Number of retries to attempt if an exception occurs before letting the exception bubble up.
+     *     @var callable $decider A callback that gets a request and an exception to decide if we should retry this or not.
+     *     @var callable $delay A callback to return how many milliseconds we should wait before trying again.
      * }
      */
     public function __construct(array $config = [])
     {
         $resolver = new OptionsResolver();
         $resolver->setDefaults([
             'retries' => 1,
+            'decider' => function (RequestInterface $request, Exception $e) {
+                return true;
```

Comment:
I do not think we can add **any** logic here because it will break the existing behavior.